### PR TITLE
DiskRP sdk changes for the swagger update of version 2020-12-01

### PR DESCRIPTION
**Pull request by meet1993shah**
# All SDK Contribution checklist:
Swagger PR - https://github.com/Azure/azure-rest-api-specs/pull/12796

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/Microsoft.Azure.Management.Compute.Tests.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.17.1-preview" />
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
     <PackageReference Include="Microsoft.Azure.Management.ManagedServiceIdentity" Version="1.0.0" />
+    <PackageReference Include="Microsoft.NETCore.Targets" Version="2.0.0" />
```
Comment (Sandido) [inline sdk/compute/Microsoft.Azure.Management.Compute/tests/Microsoft.Azure.Management.Compute.Tests.csproj:20]: Why is this needed?

**Review by Sandido — CHANGES_REQUESTED** [review]

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPCreateOptionTests.cs [hunk 2]
```diff
@@ -60,10 +60,10 @@ public void DiskFromGalleryImageVersion()
                 var rgName = TestUtilities.GenerateName(TestPrefix);
                 var diskName = TestUtilities.GenerateName(DiskNamePrefix);
                 Disk disk = GenerateBaseDisk(DiskCreateOption.FromImage);
-                disk.Location = "eastus2";
+                disk.Location = "centraluseuap";
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPCreateOptionTests.cs:63]: The galleryImage is now no longer present in the eastus2, so changed the region and image.

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPDiskEncryptionSetTests.cs [hunk 3]
```diff
@@ -35,13 +35,19 @@ public void DiskEncryptionSet_List()
         [Fact]
         public void DiskEncryptionSet_CreateDisk()
         {
-            DiskEncryptionSet_CreateDisk_Execute("DiskEncryptionSet_CreateDisk", location: "eastus2");
+            DiskEncryptionSet_CreateDisk_Execute("DiskEncryptionSet_CreateDisk", location: "centraluseuap");
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPDiskEncryptionSetTests.cs:38]: some keyencryption resources are not present in eastus2 for the tests are/were deleted, so we have changed the region to centraluseuap where the resources are present

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPDiskEncryptionSetTests.cs [hunk 4]
```diff
@@ -35,13 +35,19 @@ public void DiskEncryptionSet_List()
...
         }
 
         [Fact]
         public void DiskEncryptionSet_AddDESToExistingDisk()
         {
-            DiskEncryptionSet_UpdateDisk_Execute("DiskEncryptionSet_AddDESToExistingDisk", location: "eastus2");
+            DiskEncryptionSet_UpdateDisk_Execute("DiskEncryptionSet_AddDESToExistingDisk", location: "centraluseuap");
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPDiskEncryptionSetTests.cs:44]: some test resources are/were deleted from the eastus2, so we have changed the region to centraluseuap where the test resources are still present.

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPEncryptionTests.cs [hunk 5]
```diff
@@ -30,18 +30,18 @@ public void DiskEncryptionPositiveTest()
...
+                string secretUri = @"https://swaggerkeyvault5.vault.azure.net/secrets/swaggerSecret/0e01a1ab339e40ff9f01ea98ae8ee6b5";
 
                 string encryptionSettingsVersion = "1.0";
 
                 var rgName = TestUtilities.GenerateName(TestPrefix);
                 var diskName = TestUtilities.GenerateName(DiskNamePrefix);
                 Disk disk = GenerateDefaultDisk(DiskCreateOption.Empty, rgName, 10);
                 disk.EncryptionSettingsCollection = GetDiskEncryptionSettings(testVaultId, encryptionKeyUri, secretUri, encryptionSettingsVersion: encryptionSettingsVersion);
-                disk.Location = "eastus2";
+                disk.Location = "centraluseuap";
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/tests/DiskRPTests/DiskRPEncryptionTests.cs:44]: keyvault which is previously mentioned is deleted from the eastus2 region. so we switched to centraluseuap region where the test key vault is still present.

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/Microsoft.Azure.Management.Compute.Tests.csproj (hunk 1 above)
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/tests/Microsoft.Azure.Management.Compute.Tests.csproj:20]: will remove it

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 6]
```diff
@@ -36,5 +36,8 @@
 
   <!-- Please do not move/edit code below this line -->
   <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETCore.Targets" Version="2.0.0" />
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:40]: will remove it

File: global.json [hunk 7]
```diff
@@ -1,9 +0,0 @@
-{
-  "msbuild-sdks": {
-    "Microsoft.Build.Traversal": "1.0.45"
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply global.json]: readded

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs [hunk 8]
```diff
@@ -61,10 +61,10 @@ public PrivateEndpointConnection(PrivateLinkServiceConnectionState privateLinkSe
         partial void CustomInit();
 
         /// <summary>
-        /// Gets or sets the resource of private end point.
+        /// Gets the resource of private end point.
         /// </summary>
         [JsonProperty(PropertyName = "properties.privateEndpoint")]
-        public PrivateEndpoint PrivateEndpoint { get; set; }
+        public PrivateEndpoint PrivateEndpoint { get; private set; }
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs:67]: The property was set to read only in the swagger, thus the change

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskEncryptionSetUpdate.cs [hunk 9]
```diff
@@ -37,12 +37,17 @@ public DiskEncryptionSetUpdate()
         /// <param name="encryptionType">Possible values include:
         /// 'EncryptionAtRestWithCustomerKey',
         /// 'EncryptionAtRestWithPlatformAndCustomerKeys'</param>
+        /// <param name="rotationToLatestKeyVersionEnabled">Set this flag to
+        /// true to enable auto-updating of this disk encryption set to the
+        /// latest key version.</param>
         /// <param name="tags">Resource tags</param>
-        public DiskEncryptionSetUpdate(string encryptionType = default(string), KeyForDiskEncryptionSet activeKey = default(KeyForDiskEncryptionSet), IDictionary<string, string> tags = default(IDictionary<string, string>))
+        public DiskEncryptionSetUpdate(string encryptionType = default(string), KeyForDiskEncryptionSet activeKey = default(KeyForDiskEncryptionSet), bool? rotationToLatestKeyVersionEnabled = default(bool?), IDictionary<string, string> tags = default(IDictionary<string, string>), EncryptionSetIdentity identity = default(EncryptionSetIdentity))
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskEncryptionSetUpdate.cs:44]: added the customization

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs [hunk 10]
```diff
@@ -95,6 +99,8 @@ public DiskUpdate(OperatingSystemTypes? osType = default(OperatingSystemTypes?),
             Tier = tier;
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs:99]: customization added

**Review by markcowl — APPROVED** [review]

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (Sandido) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs:67]: @meet1993shah , @markcowl , I think this is considered a breaking change. Was it approved?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 11]
```diff
@@ -9,24 +9,17 @@
       Provides developers with libraries for the updated compute platform under Azure Resource manager to deploy virtual machine, virtual machine extensions and availability set management capabilities. Launch, restart, scale, capture and manage VMs, VM Extensions and more. Note: This client library is for Virtual Machines under Azure Resource Manager.
       Development of this library has shifted focus to the Azure Unified SDK. The future development will be focused on "Azure.ResourceManager.Compute" (https://www.nuget.org/packages/Azure.ResourceManager.Compute/). Please see the package changelog for more information.
     </Description>
-    <Version>44.0.0.0</Version>
+    <Version>45.0.0.0</Version>
```
Comment (Sandido) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:12]: Make sure to also update the file sdk/compute/Microsoft.Azure.Management.Compute/AzSdk.RP.props, similar to https://github.com/Azure/azure-sdk-for-net/pull/18498/files#diff-3d5895728d5573d745e564d34e730349416ca38f5866877e43201c05b25c3d63L4

**Review by Sandido — CHANGES_REQUESTED** [review]

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj (hunk 11 above)
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:12]: the file already has the change.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/SnapshotUpdate.cs [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+            OsType = osType;
+            DiskSizeGB = diskSizeGB;
+            EncryptionSettingsCollection = encryptionSettingsCollection;
+            Encryption = encryption;
+            NetworkAccessPolicy = networkAccessPolicy;
+            DiskAccessId = diskAccessId;
+            CustomInit();
+        }
+
+        public SnapshotUpdate(OperatingSystemTypes? osType = default(OperatingSystemTypes?), int? diskSizeGB = default(int?), EncryptionSettingsCollection encryptionSettingsCollection = default(EncryptionSettingsCollection), Encryption encryption = default(Encryption), string networkAccessPolicy = default(string), string diskAccessId = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), SnapshotSku sku = default(SnapshotSku))
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/SnapshotUpdate.cs:59]: added

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/DiskEncryptionSetUpdate.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    public partial class DiskEncryptionSetUpdate
+    {
+        /// <summary>
+        /// Initializes a new instance of the DiskEncryptionSetUpdate class.
+        /// </summary>
+        /// <param name="encryptionType">Possible values include:
+        /// 'EncryptionAtRestWithCustomerKey',
+        /// 'EncryptionAtRestWithPlatformAndCustomerKeys'</param>
+        /// <param name="activeKey">Active Key of the encryption</param>
+        /// <param name="tags">Resource tags</param>
```
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/DiskEncryptionSetUpdate.cs]: done

**Review by Sandido — DISMISSED** [review]
@meet1993shah , @markcowl , my last concern is the property that was made from Not Read Only to Read Only. That could break customers. https://github.com/Azure/azure-sdk-for-net/pull/19092#discussion_r600013340 

I think this type of change will be ok in a new version, but please include a comment in the csproj file in the changelog stating this property changed to Read Only. 
@markcowl , what do you think?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/PrivateEndpointConnection.cs:67]: the pread only change in swagger PR was approved. I will include this in the changelog.

**Comment by meet1993shah** [comment]
> @meet1993shah , @markcowl , my last concern is the property that was made from Not Read Only to Read Only. That could break customers. [#19092 (comment)](https://github.com/Azure/azure-sdk-for-net/pull/19092#discussion_r600013340)
> 
> I think this type of change will be ok in a new version, but please include a comment in the csproj file in the changelog stating this property changed to Read Only.
> @markcowl , what do you think?

Added comment in the changelog.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 14]
```diff
@@ -9,7 +9,7 @@
       Provides developers with libraries for the updated compute platform under Azure Resource manager to deploy virtual machine, virtual machine extensions and availability set management capabilities. Launch, restart, scale, capture and manage VMs, VM Extensions and more. Note: This client library is for Virtual Machines under Azure Resource Manager.
       Development of this library has shifted focus to the Azure Unified SDK. The future development will be focused on "Azure.ResourceManager.Compute" (https://www.nuget.org/packages/Azure.ResourceManager.Compute/). Please see the package changelog for more information.
     </Description>
-    <Version>45.0.0.0</Version>
+    <Version>46.0.0.0</Version>
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:12]: The latest released version is 44.0.0.0  we want this to be 45.0.0.0, including your changes and the CloudService changes

**Review by markcowl — CHANGES_REQUESTED** [review]

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj (hunk 14 above)
Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:12]: I see, okay will revert this version change. Thanks!

Comment (meet1993shah) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: reverted the change

**Comment by meet1993shah** [comment]
@markcowl , @Sandido please review again, have addressed the comment.

**Review by markcowl — APPROVED** [review]
